### PR TITLE
#47 Fix user and shovel reconcilers by setting passwordHash and srcUr…

### PR DESCRIPTION
**Pull request by bfink13**
…i. Also reconcile users before shovels

### CODE REVIEW

File: rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java [hunk 1]
```diff
@@ -83,8 +83,8 @@ public void reconcile(final Reconciliation reconciliation) throws InterruptedExc
                 deleteDanglingPvcs(resource, currentReplicaCount);
             }
 
-            shovelReconciler.reconcile(cluster);
             usersReconciler.reconcile(cluster);
+            shovelReconciler.reconcile(cluster);
```
Comment (ghost) [inline rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java:88]: Is there a reason for these to have swapped positions?  If there's a necessary order, that'd make a great comment.

File: rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java [hunk 2]
```diff
@@ -54,7 +55,10 @@ public void reconcile(final RabbitMQCluster cluster) {
                     .map(addr -> String.format("amqp://%s:%s@%s", username, password, addr.asRabbitUri()))
                     .collect(Collectors.toList());
 
-            final ShovelArguments shovelArguments = new ShovelArguments().withSrcQueue(desiredShovel.getSource().getQueue()).withDestUri(uris);
+            final ShovelArguments shovelArguments = new ShovelArguments()
+                    .withSrcUri(Lists.newArrayList("amqp://"))
```
Comment (ghost) [inline rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java]: `amqp://` seems like an odd source URI.  Is that correct?

File: rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java (hunk 1 above)
Comment (bfink13) [reply rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java:88]: added

**Review by ghost — APPROVED** [review]

File: rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java (hunk 2 above)
Comment (bfink13) [reply rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java]: yep, it just means "this cluster". moved into a const.

**Pull request merged**